swarm/network/simulation/node.go

```diff
@@ -256,8 +257,14 @@ func (s *Simulation) UploadSnapshot(snapshotFile string, opts ...AddNodeOption)
 	if err != nil {
 		return err
 	}
-	log.Info("Snapshot loaded")
-	return nil
+
+	err = s.WaitTillSnapshotRecreated(ctx, &snap)
+	if err == nil {
```

Comment:
I'm not sure this whole error check + logging is necessary. The calling context should do the error check. Personally, I'd just `return s.WaitTillSnapshotRecreated()` in this case.

Reply:
indeed. however, the logging was there in the first place, so i think i should not delete it.

Reply:
the tests that call this function already log or return the error with a fatal exception. this can be deleted IMO but as you please